### DISCUSSION TITLE
How do I access iterator index as well as value in c_foreach() for a set data type?

### DISCUSSION THREAD

**Discussion by Ashvith10** [stclib/STC#56] [General]
I want to access both the elements as well as it's index for cset_int/csset_int. The type of set doesn't matter, as long as the requirements are fulfilled.

**Comment by tylov** [comment]
```
int index = 0;
c_foreach (i, Set, set) {
   printf("%d: %d\n", index, *i.ref);
   ++index;
}
```
The index doesn't make much sense for unordered sets, but for ordered they may. If you start from the middle of the set, you will have to know at which index to start in the set if you need to retain index vs set element. (sets does not have random access like vectors).
